subprojects/gradle/runner/report/src/main/kotlin/com/avito/android/runner/report/ReportFactoryImpl.kt

```diff
@@ -0,0 +1,76 @@
+package com.avito.android.runner.report
+
+import com.avito.android.runner.report.internal.AvitoReport
+import com.avito.android.runner.report.internal.InMemoryReport
+import com.avito.android.runner.report.internal.ReportImpl
+import com.avito.http.HttpClientProvider
+import com.avito.logger.LoggerFactory
+import com.avito.report.ReportLinkGenerator
+import com.avito.report.ReportViewer
+import com.avito.report.ReportsApiFactory
+import com.avito.report.TestSuiteNameProvider
+import com.avito.time.TimeProvider
+
+public class ReportFactoryImpl(
+    private val timeProvider: TimeProvider,
+    private val useInMemoryReport: Boolean,
+    private val buildId: String,
+    private val loggerFactory: LoggerFactory,
+    private val httpClientProvider: HttpClientProvider,
+    private val reportViewerConfig: ReportViewerConfig?
```

Comment:
For me, we have a lot of places where we make decisions when `reportViewerConfig=null` or not
I want to see one place, better at the start when whole logic begins, where we check reportViewerConfig == null